x-pack/plugins/maps/public/classes/sources/es_source/es_source.ts

```diff
@@ -115,22 +195,38 @@ export class AbstractESSource extends AbstractVectorSource {
     return resp;
   }
 
-  async makeSearchSource(searchFilters, limit, initialSearchContext) {
+  async makeSearchSource(
+    searchFilters: VectorSourceRequestMeta | VectorJoinSourceRequestMeta | BoundsFilters,
+    limit: number,
+    initialSearchContext?: object
+  ): Promise<ISearchSource> {
     const indexPattern = await this.getIndexPattern();
     const isTimeAware = await this.isTimeAware();
-    const applyGlobalQuery = _.get(searchFilters, 'applyGlobalQuery', true);
-    const globalFilters = applyGlobalQuery ? searchFilters.filters : [];
-    const allFilters = [...globalFilters];
-    if (this.isFilterByMapBounds() && searchFilters.buffer) {
-      //buffer can be empty
+    const applyGlobalQuery =
+      typeof searchFilters.applyGlobalQuery === 'boolean' ? searchFilters.applyGlobalQuery : true;
+    const globalFilters: Filter[] = applyGlobalQuery ? searchFilters.filters : [];
+    const allFilters: Filter[] = [...globalFilters];
+    if (this.isFilterByMapBounds() && 'buffer' in searchFilters && searchFilters.buffer) {
+      // buffer can be empty
       const geoField = await this._getGeoField();
-      const buffer = this.isGeoGridPrecisionAware()
-        ? expandToTileBoundaries(searchFilters.buffer, searchFilters.geogridPrecision)
-        : searchFilters.buffer;
-      allFilters.push(createExtentFilter(buffer, geoField.name, geoField.type));
+      const buffer: MapExtent =
+        this.isGeoGridPrecisionAware() &&
+        'geogridPrecision' in searchFilters &&
+        typeof searchFilters.geogridPrecision === 'number'
+          ? expandToTileBoundaries(searchFilters.buffer, searchFilters.geogridPrecision)
+          : searchFilters.buffer;
+      const extentFilter = (createExtentFilter(
+        buffer,
+        geoField.name,
+        geoField.type as ES_GEO_FIELD_TYPE
+      ) as unknown) as Filter;
```

Comment:
Cast to unknown and Filter is not needed

Reply:
discussed offline. will create follow-PR. The extent-filters need to implement `Filter` correctly (https://github.com/elastic/kibana/blob/e361035650d0ea70cf36069935d53064ff3dabf5/src/plugins/data/common/es_query/filters/meta_filter.ts#L52). Out-of-scope of this PR, as it would require modifications to the filter-body.